packages/shared/src/contexts/PaymentContext.tsx

```diff
@@ -166,14 +169,21 @@ export const PaymentContextProvider = ({
     enabled: !!paddle && !!planTypes && !!geo,
   });
 
-  const productOptions: Array<ProductOption> = useMemo(
-    () =>
+  const productOptions: Array<ProductOption> = useMemo(() => {
+    const priceFormatter = new Intl.NumberFormat(
+      globalThis?.navigator?.language ?? 'en-US',
+      {
+        minimumFractionDigits: 2,
+      },
+    );
+    return (
       productPrices?.data?.details?.lineItems?.map((item) => {
         const duration = planTypes[item.price.id] as PlusPriceType;
         const priceAmount = parseFloat(item.totals.total) / 100;
-        const monthlyPrice = +(
-          priceAmount / (duration === PlusPriceType.Yearly ? 12 : 1)
-        ).toFixed(2);
+        const months = duration === PlusPriceType.Yearly ? 12 : 1;
+        const monthlyPrice = Number(
+          (priceAmount / months).toString().match(/^-?\d+(?:\.\d{0,2})?/)[0],
+        );
```

Comment:
This looks highly messy/easy to mess stuff up?

Reply:
I've tested and seems to work fine, issue with .toFixed is that is also rounding the amount and Nimrod asked me to fix it. But I'm open for another solution! 👀